enterprise/dbcrypt/cipher.go

```diff
@@ -0,0 +1,126 @@
+package dbcrypt
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/sha256"
+	"fmt"
+	"io"
+
+	"golang.org/x/xerrors"
+)
+
+type Cipher interface {
+	Encrypt([]byte) ([]byte, error)
+	Decrypt([]byte) ([]byte, error)
+	HexDigest() string
+}
+
+// CipherAES256 returns a new AES-256 cipher.
+func CipherAES256(key []byte) (*AES256, error) {
+	if len(key) != 32 {
+		return nil, xerrors.Errorf("key must be 32 bytes")
+	}
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+	aead, err := cipher.NewGCM(block)
+	if err != nil {
+		return nil, err
+	}
+	digest := fmt.Sprintf("%x", sha256.Sum256(key))[:7]
```

Comment:
Your SHA sum should include a unique string about the type of cipher, e.g. `aes256-gcm`, otherwise if we add a new type of cipher and a customer reuses a key, we'd get a collision and badness.